### ISSUE TITLE
Design grade dashboard page frontend

### ISSUE THREAD

**Issue by theyoyojo** [underground-software/singularity#66]
See all assignment grades and working final grade with relevant feedback

**Comment by theyoyojo** [comment]
json grading schema idea:

```
# cat course_fall2024.json
grading_schema = {
    "final_grade": { type: "weighted", weights: [{weight:0.2,reference:"exercises", ...}]},
    "exercises": { type: "average", entries; ["E0", "E1", "E2"],
    "programs", ...
    "midpoint", ...
    "final_project", ...
    "E0": { type: "number", location: {..where to get it..}
}
```

**Comment by theyoyojo** [comment]
includes generating list of gradable IDs for grader (initial, peer1, peer2, final)

**Comment by theyoyojo** [comment]
Fixed by #186
